tests/kernel/common/src/printk.c

```diff
@@ -17,6 +17,27 @@ int (*_old_char_out)(int);
 
 #if defined(CONFIG_PICOLIBC)
 
+#define ZEPHYR_PICOLIBC_VERSION (__PICOLIBC__ * 10000 + \
```

Comment:
So just skip tests in that case instead of verifying that they generate the expected result? I'm not sure that's as useful for this test where we're kinda walking through C library configurations and making sure they work as intended?

Reply:
Output is more or less an unexpected result with minimal io config anyway.

I guess the question is how specific do we want to be about the deviation. Fine to be overly specific as well, but just saying you have the option to be less specific too, which means PicoLibC would not be contractually bound to Zephyr to ensure minimal io formatting is a certain way. I.e. if there ever need to be modifications to picolibc's minimal io formatter, to save a few byte here or there, there is more flexibility.

Reply:
Yeah, that's a fair point. I think I'll leave this patch and burn that bridge when we get to it.